corehq/messaging/smsbackends/twilio/models.py

```diff
@@ -66,6 +85,11 @@ def send(self, msg, orig_phone_number=None, *args, **kwargs):
             if e.code == ERROR_INVALID_TO_PHONE_NUMBER:
                 msg.set_system_error(SMS.ERROR_INVALID_DESTINATION_NUMBER)
                 return
+            elif e.code == ERROR_WHATSAPP_LIMITATION:
+                notify_exception(None, "Error with Twilio Whatsapp: %s" % str(e))
+                orig_phone_number = self._convert_from_whatsapp(orig_phone_number)
+                kwargs['skip_whatsapp'] = True
```

Comment:
how about extracting out two methods
`_get_phone_number` that takes care of if the phone number needs to be modified and 
`_send_sms` that does the actual call to send and error handling?
this would avoid introducting the `skip_whatsapp` argument

Reply:
Tried this out but I still have to pass a flag around based on whether or not it's a retry, so it didn't end up being better code.

Reply:
I don't understand the rationale here, but if it's a blocker then my approval should be ignored until it's resolved.